lms/djangoapps/bulk_email/tasks.py

```diff
@@ -182,6 +182,7 @@ def perform_delegate_email_batches(entry_id, course_id, task_input, action_name)
     for qset in recipient_qsets:
         combined_set |= qset
     combined_set = combined_set.distinct()
+    combined_set = combined_set.filter(courseenrollment__course_id=course_id, courseenrollment__is_active=True)
```

Comment:
This handles the case of " a user existed in the chose cohort, but has since unenrolled from the course". Tested on my devstack to confirm.